library/ssl_tls13_generic.c

```diff
@@ -1186,6 +1134,23 @@ static int ssl_tls13_write_certificate_verify_body( mbedtls_ssl_context *ssl,
         return( MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE );
     }
 
+    ret = mbedtls_ssl_tls13_get_pk_type_and_md_alg_from_sig_alg( algorithm,
+                                                                 &pk_type,
+                                                                 &md_alg );
+    if( ret == 0 )
```

Comment:
Because of the check `mbedtls_ssl_sig_alg_is_received` just above this should not happen if there is no bug when we offer signature algorithms to the pee and check the ones proposed by the peer. Thus here I would just go `return( MBEDTLS_ERR_SSL_INTERNAL_ERROR )`.